### PR TITLE
fix(dbt): remove `dbt-core` constraint in `pyright`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Our `pyright` env has too many constraints from all its installed packages, which caused https://github.com/dagster-io/dagster/pull/21518 to fail once it was merged to master.

Relax the `dbt-core` constraint for now, but leave the enforcement for `dbt-core>=1.6` at run time.

## How I Tested These Changes
bk

### CODE REVIEW

**Comment by rexledesma** [comment]
* **#21528** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21528?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/setup.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ def get_version() -> str:
     install_requires=[
         f"dagster{pin}",
         # Follow the version support constraints for dbt Core: https://docs.getdbt.com/docs/dbt-versions/core
-        "dbt-core>=1.6,<1.8",
+        # TODO: remove dbt 1.5.x as a lower bound once the pyright environment is less constrained.
+        "dbt-core>=1.5,<1.8",
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/setup.py:41]: is there not a `pyright` specific way to resolve this? having the actual deps be inaccurate seems not ideal

Comment (alangenfeld) [reply python_modules/libraries/dagster-dbt/setup.py:41]: what happens if we remove this https://github.com/dagster-io/dagster/blob/master/pyright/master/requirements.txt#L103-L107 ?

Comment (alangenfeld) [reply python_modules/libraries/dagster-dbt/setup.py]: and https://github.com/dagster-io/dagster/blob/master/python_modules/automation/tox.ini#L64-L68

Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/setup.py]: Yeah just saw that. Removed and rebuilt the pins.

**Review by alangenfeld — APPROVED** [review]
noice, hopefully this works

**Pull request merged**